azure-functions-java-sdktypes/src/main/java/com/microsoft/azure/functions/sdktype/blob/BlobContainerHydrator.java

```diff
@@ -0,0 +1,136 @@
+package com.microsoft.azure.functions.sdktype.blob;
+
+import com.microsoft.azure.functions.sdktype.SdkTypeHydrator;
+
+import java.lang.reflect.Method;
+import java.util.logging.Logger;
+
+/**
+ * Reflection logic for building a BlobContainerClient from BlobContainerMetaData,
+ * supporting both connection string usage and managed identity usage.
+ */
+public class BlobContainerHydrator implements SdkTypeHydrator<BlobContainerMetaData> {
+    private static final Logger LOGGER = Logger.getLogger(BlobContainerHydrator.class.getName());
+
+    @Override
+    public Object createInstance(BlobContainerMetaData metaData) throws Exception {
+        String containerName = metaData.getContainerName();
```

Comment:
I am personal believer in over use of final. This is required for immutability and avoid re-assignment. Also a fact JVM optimize final variables better than mutable ones.